scripts/release.bash

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+set -eEuo pipefail
+
+# TODO; more available arguments for this script
+# instead of hardcoded values
+
+function show_help {
+    cat - <<HEREDOC
+Usage:
+
+    ${0##*/} -v VERSION [-n] [-d]
+
+    where:
+        -d            - debug mode
+        -n            - no-op mode; just print what would be done
+        RPM_LIFECYCLE - the label identyfuing repository
+                        (e.g. stable, testing, unstable). Defaults to 'dev'.
+        DEB_COMPONENT - component within the Debian repository;
+                        (e.g. stable, testing, unstable). Defaults to 'dev'.
+
+HEREDOC
+}
+
+function assert_vars_set {
+    declare n
+    for n in "$@"; do
+        if [[ -n "$n" ]] && [[ -z "${!n:-}" ]]; then
+            >&2 show_help
+            >&2 echo "ERROR: value for ${n} was not provided"
+            exit 1
+        else
+            if [[ -n ${dbgflag:-} ]]; then
+                >&2 echo "DEBUG: ${n}=${!n:-}"
+            fi
+        fi
+    done
+}
+
+declare version
+declare dbgflag=
+declare root
+declare noop=
+
+if ! command -v git >/dev/null; then
+    >&2 echo "ERROR: release of python-gcg requires git command available"
+fi
+root=$(git rev-parse --show-toplevel)
+
+while [[ $# -gt 0 ]]; do
+    declare opt="$1"
+    shift
+    case "$opt" in
+    -h)
+        show_help
+        exit 0
+        ;;
+    -d) dbgflag="-v"
+        export GIT_TRACE=1
+        ;;
+    -n) noop=echo
+        ;;
+    -v) version=${1:-}
+        shift
+        ;;
+    esac
+done
+
+if [[ -n "$dbgflag" ]]; then
+    set -x
+fi
+assert_vars_set version root
```

Comment:
If root isn't set the error will say 'a value for root was not provided', but root isn't a parameter, it comes from a git command.

Reply:
I know what you mean; considering... hmm...

- check for Git will happen earlier, and abort - so this won't show
- if the `git rev-parse` fails, script will terminate earlier as well  thanks to the bash options at the beginning:

```
$ git-changelog-generator/scripts/release.bash -v
fatal: not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

So it's merely belt and braces. At the same time yes, gut feeling tells me "don't say: this  is never going to happen"

Reply:
I've settled on renaming this parameter to something more meaningful (`git_repo_rootdir`)